src/v/cluster/feature_manager.h

```diff
@@ -177,6 +186,9 @@ class feature_manager {
     // Keep track of whether this node is the controller leader
     // via leadership notifications
     bool _am_controller_leader{false};
+
+    // Blocks cluster upgrades until the enterprise license has been verified
+    bool _verified_enterprise_license{false};
```

Comment:
suggestion: could we just as easily use a semaphore for this? do an abortable sem::wait rather than explicitly sleep in a loop?

Reply:
Done

Reply:
i think in this case we can use a `ss::condition_variable` it is intended to be used as a `wait()` `notify()` mechanism.